### PR TITLE
[WIP] reports for import definitions processes

### PR DESCRIPTION
**Pull request by wpeisert**

### CODE REVIEW

File: src/ProcessManagerBundle/Controller/ProcessReportController.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class ProcessReportController extends AdminController
+{
+    /**
+     * @Route("/admin/process_manager/reports/log-download/{id}")
+     * @return BinaryFileResponse
+     * @return JsonResponse
+     */
+    public function LogDownloadAction($id)
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Controller/ProcessReportController.php]: camel case, lowercase 'L' please

File: src/ProcessManagerBundle/Resources/config/pimcore/routing.yml [hunk 2]
```diff
@@ -16,4 +16,9 @@ process_manager_admin_processes:
   type: coreshop.resources
   resource: |
     alias: process_manager.process
-    only: ['list', 'delete']
\ No newline at end of file
+    only: ['list', 'delete']
+
+process_manager_admin_processes_report:
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Resources/config/pimcore/routing.yml:21]: don't use annotations for routing please

File: src/ProcessManagerBundle/Controller/ProcessReportController.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        $response = new BinaryFileResponse($filePath);
+        $response->headers->set('Content-Type', 'text/plain');
+        $response->setContentDisposition(ResponseHeaderBag::DISPOSITION_ATTACHMENT);
+
+        return $response;
+    }
+
+    /**
+     * @Route("/admin/process_manager/reports/get")
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Controller/ProcessReportController.php:60]: don't use annotations for routing please

File: src/ProcessManagerBundle/Controller/ProcessReportController.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+use Symfony\Component\HttpFoundation\BinaryFileResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class ProcessReportController extends AdminController
+{
+    /**
+     * @Route("/admin/process_manager/reports/log-download/{id}")
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Controller/ProcessReportController.php]: don't use annotations for routing please

File: src/ProcessManagerBundle/Tool/Report.php [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+/**
+ * Class Report
+ *
+ * Prepares import process report from log file
+ *
+ * @package ProcessManagerBundle\Tool
+ */
+class Report
+{
+    const SKIP_LINES_SHORTER_THAN = 50;
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Tool/Report.php]: maybe a config?

File: src/ProcessManagerBundle/Tool/Report.php [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+        $link = '/admin/process_manager/reports/log-download/' . $this->processId;
+        $items[] = "Log file: <a href=\"$link\" target=\"_blank\">download</a>";
+
+        if (count($this->errors)) {
+            $items[] = 'Errors: ';
+            $items = array_merge($items, $this->errors);
+        }
+
+        return implode('<br />', $items);
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Tool/Report.php]: Twig Rendering?

File: src/ProcessManagerBundle/Controller/ProcessReportController.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $success = true;
+        $html    = '';
+        $message = '';
+
+        $id = $request->get('id');
+
+        try {
+            $report = new Report($id);
+            $html = $report->getReportHtml();
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Controller/ProcessReportController.php]: don't use a model for business logic, use a service instead.

**Review by dpfaffenbauer — CHANGES_REQUESTED** [review]
Idea is very good but the implementation needs refactoring.

File: src/ProcessManagerBundle/Tool/Report.php [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+ *
+ * Prepares import process report from log file
+ *
+ * @package ProcessManagerBundle\Tool
+ */
+class Report
+{
+    const SKIP_LINES_SHORTER_THAN = 50;
+
+    const EVENT_TOTAL = 'import_definition.total:';
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Tool/Report.php]: Logging doesn't happen here, ImportDefinitions should use a Service from ProcessManager for Logging

**Review by dpfaffenbauer — CHANGES_REQUESTED** [review]

File: src/ProcessManagerBundle/Controller/ProcessReportController.php (hunk 1 above)
Comment (wpeisert) [reply src/ProcessManagerBundle/Controller/ProcessReportController.php]: sure

File: src/ProcessManagerBundle/Resources/config/pimcore/routing.yml (hunk 2 above)
Comment (wpeisert) [reply src/ProcessManagerBundle/Resources/config/pimcore/routing.yml:21]: I'm not sure how to change it. Used here routing syntax is not clear for me.

File: src/ProcessManagerBundle/Tool/Report.php (hunk 5 above)
Comment (wpeisert) [reply src/ProcessManagerBundle/Tool/Report.php]: This is a technical value just to skip garbage.

File: src/ProcessManagerBundle/Tool/Report.php (hunk 6 above)
Comment (wpeisert) [reply src/ProcessManagerBundle/Tool/Report.php]: I would rather postpone the feature

File: src/ProcessManagerBundle/Service/ProcessLog.php [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+{
+    const LOG_DIRECTORY = 'process_log/';
+    const LOG_PREFIX = 'process_';
+    const LOG_EXTENSION = 'log';
+
+    /**
+     * @param int $processId
+     * @param string $message
+     */
+    public function logEvent($processId, $message)
```
Comment (dpfaffenbauer) [inline src/ProcessManagerBundle/Service/ProcessLog.php:26]: monolog maybe?

File: src/ProcessManagerBundle/Tool/Report.php (hunk 6 above)
Comment (dpfaffenbauer) [reply src/ProcessManagerBundle/Tool/Report.php]: ok

**Comment by dpfaffenbauer** [comment]
Seems like I can't push to your PR pretty easily... Here is my commit I created. This is a bit more advanced than your proposed solution. My solution uses MonoLog and a Handler Factory that can be created for each type -> so you have the possibility to log where ever you want. Also the Report Class is a separate service so you can have different types of reports based on the process type:

https://github.com/dpfaffenbauer/ProcessManager/commit/164c9c223d12aa3309eda6770724ba35b74d4549

**Comment by dpfaffenbauer** [comment]
Better look at that, full diff here: 

https://github.com/dpfaffenbauer/ProcessManager/compare/wpeisert-master?expand=1

Here are the changes for ImportDefinitions:

https://github.com/w-vision/ImportDefinitions/compare/process-manager-changes?expand=1

**Pull request merged**